features/step_definitions/cli.rb

```diff
@@ -236,17 +236,19 @@
     raise "Failed to patch #{resource_type} #{resource_name} with #{patch_json}"
   end
 
-  opts = {resource: resource_type, resource_name: resource_name, o: "yaml", _quiet: true}
-  sec = 30
-  failpath = nil
-  success = wait_for(sec) {
-    failpath = []
-    res = _user.cli_exec(:get, **opts)
-    substruct?(expect_hash, res[:parsed], vague_nulls: true, failpath: failpath, exact_arrays: true, null_deletes_key: true)
-  }
-
-  if ! success
-    raise "patch failed to apply at #{failpath}! #{@result[:response]}"
+  if patch_type != "json" # below substruct? does not apply for json patch
```

Comment:
Could you test scenario with steps like `Given I successfully patch resource "pod/hooks-1-deploy" with:`
The patch_type is optional is this situation.

Reply:
No need to test, because if is optional, it is still != "json".
But I do a test, anyway: http://pastebin.test.redhat.com/970644 against https://github.com/xingxingxia/cucushift/blob/test-only/mytest.feature with `And I successfully patch resource "configmap/myconfigmap" with:`, no issue.